docs/recategorizing-applications.md

```diff
@@ -0,0 +1,286 @@
+categorize them manually given their request paths. For example, if you know
+the request path `/node/[host]/[port]/foo/bar` matches the `foo` application,
+you may wish to add a filter that assigns the `foo` application to any page
+impressions with that request path, and apply that filter to all the previously
+ingested page impressions without having to reingest the original log files.
+
+To help with recategorization, as page impressions are ingested, their distinct
+request paths are kept in the `modw_ondemand.request_path` database table, with
+a `request_path_id` foreign key in the `modw_ondemand.page_impressions` fact
+table. These can be used to recategorize previously ingested page
+impressions by selecting all the page impressions that have certain request
+paths and reassigning an application to them (instructions for doing this are
+further down).
+
+In the cases where multiple distinct request paths can be grouped together in a
+general form without losing any relevant information (e.g., `/foo?a=b`,
+`/foo?c=d`, and `/foo?e=f` might all be generalized as `/foo?[params]`), there
+is a configuration file that filters request paths into general forms such that
+they appear in the `modw_ondemand.request_path` table as, e.g.,
+`/foo?[params]`. You may wish to recategorize groups of request paths into a
+more general form to keep the size of this table smaller and to simplify the
+query for recategorizing applications of a general form (instructions for
+doing this are immediately below).
+
+## Recategorizing future request paths
+
+The file used for filtering request paths into more general forms is in the
+configuration directory (whose location depends on how you configured your
+installation, e.g., `/etc/xdmod`, `/opt/xdmod/etc`) under
+`etl/etl_data.d/ood/request-path-filter.json`. This file contains a JSON object
+whose keys are regular expressions used for matching against request paths from
+the ingested logs, and whose values are the corresponding general forms that
+will be stored in the database and assigned to future page impressions when
+they are ingested. Edit this file to configure the filter how you wish.
+
+If you have multiple OnDemand resources configured in XDMoD, you can also
+filter request paths on a per-resource basis by having other similar files at
+`etl/etl_data.d/ood/request-path-filter.d/${OOD_RESOURCE_CODE}.json`, where
+`${OOD_RESOURCE_CODE}` matches the value passed via the `-r` argument to
+`xdmod-ondemand-ingestor` (see the [ingestion and aggregation
+instructions](ingestion-aggregation.md)). Any keys that are the same between
+these files and the main `request-path-filter.json` will have their values
+override the values in the main `request-path-filter.json`.
+
+## Recategorizing existing request paths
+
+To apply a new filter to page impressions that have already been ingested,
+follow the instructions below to run some SQL statements in the XDMoD data
+warehouse.
+
```

Comment:
The first step is to make a backup of the database. - please can you add text that says this (and which tables / schemas should be backedup)